src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SQLFunctionsIT.java

```diff
@@ -316,6 +318,149 @@ public void castIntFieldToDoubleWithAliasJdbcFormatGroupByTest() {
         }
     }
 
+    @Test
+    public void castKeywordFieldToDatetimeWithoutAliasJdbcFormatTest() {
+        JSONObject response = executeJdbcRequest("SELECT CAST(date_keyword AS DATETIME) FROM "
+                + TestsConstants.TEST_INDEX_DATE + " ORDER BY date_keyword");
+
+        String date_type_cast = "{\"name\":\"cast_date_keyword\",\"type\":\"date\"}";
+        assertEquals(response.getJSONArray("schema").get(0).toString(), date_type_cast);
+        String[] expectedOutput = new String[] {"Tue Aug 19 07:09:13", "Wed Sep 25 02:04:13"};
+
+        for (int i = 0; i < response.getJSONArray("datarows").length(); ++i) {
```

Comment:
The assert in for-loop may not executed when the array is empty. Refer the implementation of https://github.com/opendistro-for-elasticsearch/sql/blob/3730ac6e9b3a363c51432b34d2104567eb9babb7/src/test/java/com/amazon/opendistroforelasticsearch/sql/util/MatcherUtils.java#L144